src/main/java/decodes/tsdb/algo/PythonAlgorithm.java

```diff
@@ -310,6 +312,14 @@ protected void beforeTimeSlices()
 			}
 			catch(Exception ex)
 			{
+				if (ex instanceof PyException) {
```

Comment:
Would it make more sense to just catch PyException directly?

Reply:
I see a couple ways this should be improved, thanks!